### PR TITLE
[Feat]: Setting up Prompts for release workflow

### PR DESCRIPTION
**Pull request by agrittiwari**
In this PR:

Added Release workflow with expo-updates,
Wrote Scripts to manage releases across 3 configs/environments, namely development, staging and production
Callback to update OTA changes on entry point file `App.tsx`

Scope:
The scope of this feature is basically setup whole Release worflow, another way is codepush, rn-cli folks use this heavily

And one can also include workspace related configurations , github actions etc.

### CODE REVIEW

**Comment by agrittiwari** [comment]
@danstepanov lmk wdyt

**Comment by danstepanov** [comment]
@agrittiwari what is the specific feedback you're asking for? Also can you indicate which changes pertain to your PR vs just the formatting stuff?

**Comment by agrittiwari** [comment]
@danstepanov  Need some review on the PR, specifically regarding this, "should I ask for project Id in prompt or left an info in cli saying, " update, projectId in app.json`

**Comment by agrittiwari** [comment]
> @agrittiwari what is the specific feedback you're asking for? Also can you indicate which changes pertain to your PR vs just the formatting stuff?

II want to know how to remove the formatting related changes from the PR!!!

File: src/utilities/runCLI.ts [hunk 1]
```diff
@@ -58,48 +58,94 @@ export async function runCLI(toolbox: Toolbox): Promise<CliResults> {
...
+
+  const isEAScliInstalled = await confirm(
+    'Do you have EAS cli installed ? ',
+    true
+  )
+
+  const { releaseWorkflowSetup } = await ask(askReleaseWorkflowSetup)
+
+  if (releaseWorkflowSetup === 'none') {
+    success(`Ohkay!!, so you will build apks manually with gradle and xCode.`)
```
Comment (danstepanov) [inline src/utilities/runCLI.ts:131]:
```suggestion
    success(`Sounds good 👍🏻`)
```

File: src/utilities/runCLI.ts [hunk 2]
```diff
@@ -58,48 +58,94 @@ export async function runCLI(toolbox: Toolbox): Promise<CliResults> {
...
+  } else if (releaseWorkflowSetup === 'EAS') {
```
Comment (danstepanov) [inline src/utilities/runCLI.ts:132]:
```suggestion
  } else {
```

File: src/utilities/runCLI.ts [hunk 3]
```diff
@@ -58,48 +58,94 @@ export async function runCLI(toolbox: Toolbox): Promise<CliResults> {
...
+      })
+      success(
+        ` Great! EAS will set your environments, namely development, staging and production , and you can share your app and send OTA updates to any specific environment`
+      )
+    }else{
+      error(`Need to install EAS cli to setup EAS release workflow`)
+    }
+         
+  } else if (releaseWorkflowSetup === 'Codepush') {
+    success(`Great you chose to setup Codepush updates, but it's not supported right now`)
```
Comment (danstepanov) [inline src/utilities/runCLI.ts:147]: you can remove this

**Review by danstepanov — CHANGES_REQUESTED** [review]
I'm not sure this will work if the user chooses Expo Router as `App.tsx` doesn't exist for Expo Router. Can you add support for that case?

**Review by frankcalise — COMMENTED** [review]
So I don't know if it matters, but releasing (EAS Build) and OTA Updates (expo-updates package plus EAS update) can be considered two different things.

If you want to scale to CodePush later on as an option, you likely still want EAS Build configuration to happen (in fact I'd probably just support this no matter what the OTA strategy is), but not some of the other pieces such as:
- The `eas.json` just wouldn't need to know about the `channel` properties
- `app.json` wouldn't need the `updates` key

Both of those are pertaining to `expo-updates`

File: src/templates/base/app.json.ejs [hunk 4]
```diff
@@ -27,6 +27,13 @@
       "resizeMode": "contain",
       "backgroundColor": "#ffffff"
     },
+    <% if (props.releasePackage?.name === 'expo-updates') { %>
+      "updates": {
```
[the remark this answers is not in the document]
Comment (AGRIT420) [reply src/templates/base/app.json.ejs:31]: This is following this doc , added  checkAutomatically option as well
[https://docs.expo.dev/versions/latest/config/app/#fallbacktocachetimeout](https://docs.expo.dev/versions/latest/config/app/#fallbacktocachetimeout)

File: src/utilities/runCLI.ts [hunk 5]
```diff
@@ -58,48 +58,94 @@ export async function runCLI(toolbox: Toolbox): Promise<CliResults> {
...
+       cliResults.packages.push({
+        name: 'expo-updates',
+        type: 'releaseWorkflow',
+        options: {},
+      })
+      success(
+        ` Great! EAS will set your environments, namely development, staging and production , and you can share your app and send OTA updates to any specific environment`
+      )
+    }else{
+      error(`Need to install EAS cli to setup EAS release workflow`)
```
[the remark this answers is not in the document]
Comment (AGRIT420) [reply src/utilities/runCLI.ts:143]: on the user's machine, I can't be added in dependencies, user has to login to their eas profile. It's equivalent to github-cli

**Comment by danstepanov** [comment]
EAS will be part of the paid starter, closing this for now

**Pull request closed without merging**